brave/src/main/java/com/linecorp/armeria/client/brave/BraveClient.java

```diff
@@ -0,0 +1,68 @@
+/*
+ * Copyright 2019 LINE Corporation
+ *
+ * LINE Corporation licenses this file to you under the Apache License,
+ * version 2.0 (the "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at:
+ *
+ *   https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+
+package com.linecorp.armeria.client.brave;
+
+import static com.linecorp.armeria.common.brave.RequestContextCurrentTraceContext.ensureScopeUsesRequestContext;
+
+import java.util.function.Function;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.linecorp.armeria.client.Client;
+import com.linecorp.armeria.client.tracing.HttpTracingClient;
+import com.linecorp.armeria.common.HttpRequest;
+import com.linecorp.armeria.common.HttpResponse;
+import com.linecorp.armeria.common.brave.RequestContextCurrentTraceContext;
+
+import brave.Tracing;
+import brave.http.HttpTracing;
+
+/**
+ * Decorates a {@link Client} to trace outbound {@link HttpRequest}s using
+ * <a href="https://github.com/openzipkin/brave">Brave</a>.
+ *
+ * <p>This decorator puts trace data into HTTP headers. The specifications of header names and its values
+ * correspond to <a href="http://zipkin.io/">Zipkin</a>.
+ */
+public final class BraveClient extends HttpTracingClient {
+
+    private static final Logger logger = LoggerFactory.getLogger(BraveClient.class);
+
+    /**
+     * Creates a new tracing {@link Client} decorator using the specified {@link Tracing} instance.
+     */
+    public static Function<Client<HttpRequest, HttpResponse>, BraveClient> newDecorator(
+            HttpTracing httpTracing) {
+        try {
+            ensureScopeUsesRequestContext(httpTracing.tracing());
+        } catch (IllegalStateException e) {
+            logger.warn("{} - it is appropriate to ignore this warning if this client is not being used " +
+                        "inside an Armeria server (e.g., this is a normal spring-mvc tomcat server).",
+                        e.getMessage());
+        }
+        return delegate -> new BraveClient(delegate, httpTracing);
+    }
+
+    /**
+     * Creates a new instance.
+     */
+    private BraveClient(Client<HttpRequest, HttpResponse> delegate, HttpTracing httpTracing) {
+        super(delegate, httpTracing.tracing(), httpTracing.serverName(),
```

Comment:
Since we aren't migrating to use the parser and other fields of `HttpTracing`, and it'd be a pretty big change, I think we should stick to accepting `Tracing` for now like before. I think even long-term, it's reasonable to expose an API that takes `Tracing` and one that takes `HttpTracing` later.

If we did stick with accepting `HttpTracing`, we need to add to the javadoc that we currently do not respect all the fields in it

Reply:
I agree. :-) Let me remove the `HttpTracing` for now. Will just leave a `TODO` comment for that. 😉